OpenRA.Mods.Common/Traits/Render/Hovers.cs

```diff
@@ -43,7 +43,7 @@ public IEnumerable<IRenderable> ModifyRender(Actor self, WorldRenderer wr, IEnum
 			if (self.World.Paused || IsTraitDisabled)
 				return r;
 
-			var visualOffset = !aircraft || self.CenterPosition.Z > 0 ? (int)Math.Abs((self.ActorID + Game.LocalTick) / 5 % 4 - 1) - 1 : 0;
+			var visualOffset = !aircraft || self.World.Map.DistanceAboveTerrain(self.CenterPosition).Length > 0 ? (int)Math.Abs((self.ActorID + Game.LocalTick) / 5 % 4 - 1) - 1 : 0;
```

Comment:
Thanks, I missed that. Thought I would not need to make check my oneline changes :P